nncf/data/generators.py

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) 2024 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, TypeVar
+
+import nncf
+
+BASE_VOCAB_SIZE = 12000
+
+TModel = TypeVar("TModel")
+TTokenizer = TypeVar("TTokenizer")
+
+
+def generate_text_data(
+    model: TModel, tokenizer: TTokenizer, seq_len: int = 32, dataset_size: int = 128, unique_tokens_lower_limit: int = 5
+) -> List[str]:
```

Comment:
@AlexKoff88 asked me to modify this in a few comments below:

> I missed that part during the last review but I would like to name the function generate_text_data(model: PreTrainedModel, tokenizer: PreTrainedTokenizer, max_length: int = 32,  num_samples: int = 128) -> List[str]

I changed as was asked. For now, you're requesting to roll back this change. Please, discuss this with @AlexKoff88 and provide your final decision on that.

Reply:
@AlexKoff88 , can you please describe the use case with some example so that it's clear how the new API will be used? I see that there is some misunderstanding.

Reply:
@alexsu52, are there any updates on that? What is the final decision?